### PR TITLE
Added number of subsets with given diff in dp in c++

### LINKED ISSUE #2985: Count of subsets with given difference in c++ using Dynamic Programming
## 🚀 Feature
I am GSSOC'21 participant and would like to add this problem of counting subsets with the given difference in c++ using Dynamic Programming.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## Motivation

(Please outline the motivation for the proposal.)
Since everyone thinks that Dynamic programming is very tough, so I want to give a simple approach to this problem which seems to be tough

## Pitch

(Please explain why this feature should be implemented and how it would be used.)
This problem can be easily implemented by using the problem that is the count of subsets with a given sum.

### PR DESCRIPTION
**Pull request by shriya030**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added number of subsets with given difference in dp in c++

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixed #2985

### CODE REVIEW

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code

**Comment by shriya030** [comment]
Changes done!

File: C-Plus-Plus/dp/numberOf_subsets_with_given_diff.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
+// Program to find number of subsets with given difference between their sums
+// Uses dynamic programming to solve the problem. We'll use the existing problem that counts subsets with a given sum.
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/numberOf_subsets_with_given_diff.cpp]: - Make your file name concise and readable `numberOf_subsets_with_given_diff.cpp`
- Add a beginner-friendly description at the top 
- Add a new line after description

File: C-Plus-Plus/dp/numberOf_subsets_with_given_diff.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+#include <bits/stdc++.h>
+using namespace std;
+int dp[20001][20001];
+// counting number of subsets
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/numberOf_subsets_with_given_diff.cpp]: Remove that redundant comments

File: C-Plus-Plus/dp/numberOf_subsets_with_given_diff.cpp [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+{
+	int arr_sum = 0;
+	for (int i = 0; i < n; i++)
+		arr_sum += arr[i];
+	/*Let there be two subsets with sum s1 and s2 respectively
+	s1 + s2 = arr_sum
+	s1 - s2 = diff
+	so solving the above equations, we get:
+	s1 = (diff + arr_sum)/2
+	Therefore, we will find number of subsets with sum as s1 */
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/numberOf_subsets_with_given_diff.cpp]: Remove that redundant comments

File: C-Plus-Plus/dp/numberOf_subsets_with_given_diff.cpp [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	int sum = (diff + arr_sum) / 2;
+	return countSubsets(arr, n, sum);
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/numberOf_subsets_with_given_diff.cpp]: Add a newline above the return statement.

File: C-Plus-Plus/dp/numberOf_subsets_with_given_diff.cpp [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+// Explanation:
+// Three pairs of subsets:
+// 1.[1, 1, 2] and[3]
+// 2.[1, 2] and[1, 3]
+// 3.[1, 3] and[1, 2] (Because there are two 1's)
+
+// Complexity Analysis: 
+// Time Complexity: O(sum*n), where sum is the ‘sum’ in the function subsetsDiff and ‘n’ is the size of array.
+// Auxiliary Space: O(sum*n), as the size of 2-D array is sum*n.
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/numberOf_subsets_with_given_diff.cpp]: - Add sample I/O at the end of the Code.
-Your Input and Output are not understandable. How you're reading I/P's and What it is producing as O/P?

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/dp/numberOf_subsets_with_given_diff.cpp (hunk 3 above)
Comment (shriya030) [reply C-Plus-Plus/dp/numberOf_subsets_with_given_diff.cpp]: This comment explains the main logic of the code which is making it simpler. Should I remove this?

Comment (ankitaggarwal23) [reply C-Plus-Plus/dp/numberOf_subsets_with_given_diff.cpp]: Okay then no need to remove. Use multiline comments. Don't use single line comments.

**Review by ankitaggarwal23 — COMMENTED** [review]
Just Do the remaining changes.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**